lib/card/set_pattern.rb

```diff
@@ -77,6 +77,15 @@ def initialize card
         else
           Card.fetch_id @anchor_name
         end
+        if self.class.pattern_code == 'type' and
+             default_rule = Card.fetch(card.type_cardname.trait_name(:type).trait_name(:default),
+                                       :skip_modules=>true, :skip_virtual=>true) and
+             type_code = default_rule.type_code and
+             mod_key = "Type::#{type_code.to_s.camelize}" and
+             (Card::Set.modules[:nonbase_format].values + [Card::Set.modules[:nonbase]]).
+               any?{|hash| hash[mod_key]}
+          @inherited_type_module_key = mod_key
```

Comment:
would be nice to put this is a well-named method and make it more readable.   This is a lot to tuck inside initialization with no explanation or clean naming.

actually, hold on.  there are a couple of ways to improve this:
1.  Override initialize in Card::TypeSet (can be in this file).  then you don't have to check the pattern_code
2. there should be a method for getting the set cardname for these instances, which are basically sets.

I like this quote: https://github.com/bbatsov/ruby-style-guide#naming